Makefile

```diff
@@ -382,36 +399,49 @@ $(TEST_OUTPUT_ROOT):
 
 prepare-coverage-test: $(GOTESTSUM) $(TEST_OUTPUT_ROOT)
 
+# Runs gotestsum, retrying failed tests as specified by the FAILED_TEST_RETRIES variable.
+# A function is required here there are two variants of this command, putting the packages to test in different
+# positions.
+# NOTE: Packages need to be provided via a named argument when --rerun-fails is on as stated in the gotestsum README.
+# NOTE: This had to be a long line to avoid adding newlines to the generated command.
+define gotestsum_coverage
+	$(if $(shell test $(FAILED_TEST_RETRIES) -eq 0 && echo "OK"), $(GOTESTSUM) --junitfile $(NEW_REPORT) -- $(1), $(GOTESTSUM) --rerun-fails=$(FAILED_TEST_RETRIES) --rerun-fails-max-failures=10 --junitfile $(NEW_REPORT) --packages $(1) --)
```

Comment:
first, can we split this onto a few lines for readability? I see the comment but there's got to be a way to do it with backslashes or something.

second, using shell to check a make/env variable seems unnecessary. why not use make's `ifeq` (with two versions of the function in the two branches)?

third, why not just use `--packages` all the time? isn't that simpler?

Reply:
I thought `--packages` was incompatible with `TEST_ARGS="-run ..."` but I was wrong. Simplified.